### PR TITLE
Update PckgCI cpu testing to include SDXL model testing

### PR DESCRIPTION
**Pull request by saienduri**
ci-exactly: build_packages, regression_test_cpu, regression_test_amdgpu_vulkan, regression_test_amdgpu_rocm, regression_test_nvidiagpu_vulkan, regression_test_nvidiagpu_cuda

### CODE REVIEW

File: .github/workflows/pkgci_regression_test_cpu.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
         uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
         with:
           repository: nod-ai/SHARK-TestSuite
-          ref: e67779cee2a3878c2ba9ede50f2121cf21c1e99c
+          ref: 1d77dcffdffc658e4defa6021a60ca1e9d065c61
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_cpu.yml]: Please update the other workflows to also use the same ref. I'd like to keep them all running the same code.

https://github.com/iree-org/iree/blob/cda70e8a3e235baf50b7da31c8df75d5db3d56a9/.github/workflows/pkgci_regression_test_amdgpu_rocm.yml#L62-L63 https://github.com/iree-org/iree/blob/cda70e8a3e235baf50b7da31c8df75d5db3d56a9/.github/workflows/pkgci_regression_test_amdgpu_vulkan.yml#L59-L60 https://github.com/iree-org/iree/blob/cda70e8a3e235baf50b7da31c8df75d5db3d56a9/.github/workflows/pkgci_regression_test_nvidiagpu_cuda.yml#L59-L60 https://github.com/iree-org/iree/blob/cda70e8a3e235baf50b7da31c8df75d5db3d56a9/.github/workflows/pkgci_regression_test_nvidiagpu_vulkan.yml#L59-L60

(I just find/replace across the repo from the previous commit SHA to the new one)

File: .github/workflows/pkgci_regression_test_cpu.yml [hunk 2]
```diff
@@ -71,14 +71,58 @@ jobs:
...
-          python SHARK-TestSuite/iree_tests/download_remote_files.py
+          python SHARK-TestSuite/iree_tests/download_remote_files.py --root-dir pytorch/models
       - name: Run external tests - Models with real weights
         run: |
           source ${VENV_DIR}/bin/activate
-          pytest SHARK-TestSuite/iree_tests/pytorch/ -k real_weights \
+          pytest SHARK-TestSuite/iree_tests/pytorch/models -k real_weights \
                -rpfE --timeout=600 --durations=0 \
-              --config-files=build_tools/pkgci/external_test_suite/pytorch_cpu_llvm_task.json
+              --config-files=SHARK-TestSuite/iree_tests/configs/config_pytorch_models_cpu_llvm_task.json
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_cpu.yml]: Let's add a new config file here in the IREE repo (https://github.com/iree-org/iree/tree/main/build_tools/pkgci/external_test_suite) so developers can edit flags and XFAIL lists as the code in IREE changes.

* The test suite repo contains test sources and common test runner configuration code
* Downstream repos decide which of those tests to run, with which flags, on which devices

File: .github/workflows/pkgci_regression_test_cpu.yml [hunk 3]
```diff
@@ -71,14 +71,58 @@ jobs:
           pytest SHARK-TestSuite/iree_tests/onnx/ \
               -n auto -rpfE --timeout=30 --retries=2 --durations=20 \
               --config-files=build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json
-      # TODO(scotttodd): add a local cache for these large files to a persistent runner
+  # persistent runner with cache enabled for model testing
+  linux_x86_64_models:
+    name: Linux (x86_64) Model Testing/Benchmark
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_cpu.yml:77]: I'm keeping an eye on workflow / job names. This is rather long:

`PkgCI / Regression Test CPU / Linux (x86_64) Model Testing/Benchmark (pull_request)`

May do a scrub through the various workflow files and pick new names at some point.

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: .github/workflows/pkgci_regression_test_cpu.yml (hunk 1 above)
Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test_cpu.yml]: If you bump to the latest commit, you can also add `--no-skip-tests-missing-files` to the pytest flags

File: .github/workflows/pkgci_regression_test_cpu.yml [hunk 4]
```diff
@@ -71,14 +71,58 @@ jobs:
...
           source ${VENV_DIR}/bin/activate
-          pytest SHARK-TestSuite/iree_tests/pytorch/ -k real_weights \
+          pytest SHARK-TestSuite/iree_tests/pytorch/models -k real_weights \
                -rpfE --timeout=600 --durations=0 \
-              --config-files=build_tools/pkgci/external_test_suite/pytorch_cpu_llvm_task.json
+              --config-files=SHARK-TestSuite/iree_tests/configs/config_pytorch_models_cpu_llvm_task.json
+      - name: "Running SDXL pipeline benchmark"
+        run: |
+          source ${VENV_DIR}/bin/activate
+          bash SHARK-TestSuite/iree_tests/benchmarks/benchmark_sdxl_cpu.sh
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test_cpu.yml]: May want to put benchmarking in a separate job so we can choose to run benchmarks less frequently than tests (e.g. only on postsubmit or when a PR author opts in). Looking at the logs at https://github.com/iree-org/iree/actions/runs/8760237017/job/24045123320?pr=17117, testing takes ~6 minutes and benchmarking takes ~5 minutes)

Also did this only run one iteration? Benchmarks should run at least a few iterations to ensure stable results:
```
----------------------------------------------------------------------------------------------------
Benchmark                                          Time             CPU   Iterations UserCounters...
----------------------------------------------------------------------------------------------------
BM_tokens_to_image/process_time/real_time     282651 ms      2175540 ms            1 items_per_second=3.53793m/s
```

(separately we'll want to ensure that benchmark actually makes sense... fp16 on CPU isn't expected to be fast)

---

Maybe we should start with
* tests on CPU
* tests on GPU
* benchmarks on GPU

then add CPU benchmarks later...?

Comment (saienduri) [reply .github/workflows/pkgci_regression_test_cpu.yml]: Sure sgtm. Removed the cpu benchmarks for now and addressed all the other comments as well. Should be good to go

**Review by ScottTodd — APPROVED** [review]
Nice!

File: .github/workflows/pkgci_regression_test_cpu.yml (hunk 4 above)
Comment (ScottTodd) [reply .github/workflows/pkgci_regression_test_cpu.yml]: Can you update the PR title + description to clarify that benchmarks aren't included in this initial PR?

**Pull request merged**